### DISCUSSION TITLE
Can I import a password-store from external storage?

### DISCUSSION THREAD

**Discussion by danielittlewood0** [zeapo/Android-Password-Store#1406] [Q&A]
I am trying to migrate between two devices, which each have Password Store installed.
The old store is completely local. I don't have it hosted on a git server anywhere.
I have exported the old store to a folder using the app, and moved it to the internal storage of the new device.
By setting up an "External storage" repo, I can use my passwords on the new device.
Ideally, I would be able to import the old passwords into the app's Hidden Storage on my new device. Is this possible?

**Comment by msfjarvis** [comment]
It is not possible because once your store is in the app's hidden directory, you will not be able to perform any kind of sync on it. That is why we recommend the use of hidden storage for Git users, and external storage for users who prefer to sync the store using a different solution such as Nextcloud or Syncthing.

**Comment by oblitum** [comment]
@msfjarvis hi.

> and external storage for users who prefer to sync the store using a different solution such as Nextcloud or Syncthing.

I wish to use the android app without any involvement of git, as I'm syncing my stuff with a different solution, is it possible? I tried to use the android app but all the interface seems towards having a git repo or having to clone one.

**Reply by msfjarvis** [reply]
The app has an explicit setup flow that does not enforce Git, through the "Create local repo -> SD Card" options in the onboarding flow.

**Reply by oblitum** [reply]
@msfjarvis Hi. I tried that, but in the end of the process I got nothing in my  app's main page. Given it was empty, I tried to drag down to trigger a refresh, when I did that a popup showed up stating a repository url had to be set first. Despite that, I didn't find it that quite explicit actually, because the way the app behaves seems to indicate it's expecting a local **git** repository that's saved in the SD Card, at least that's the impression it gives me. It should use wording like "local password store" instead of "repository" etc, and never refer to repo url, etc when that route is taken, to avoid the confusion.

**Reply by msfjarvis** [reply]
The public release is fairly outdated compared to our development branch and will stay that way for a while. You'll find that our [snapshot builds](https://android-password-store.github.io/docs/users/release-channels) offer a markedly better experience, which I've personally verified right before posting this comment.

**Reply by oblitum** [reply]
Ah okay, thx, I'll try checking that.

**Reply by oblitum** [reply]
@msfjarvis just tried that and ended up the same: empty main page. All I did was to follow initial app walk-through, I've set a local password store path on my SD card, it asked for key access on OpenKeychain, which I selected, after that no passwords on front page, refresh of it asks for cloning repository first, `...` at top right offers "Synchronize repository" which gives a "Git url is not set!" error, there's also "Pull/Push from remote" which doesn't make sense for a bare _gitless_ local store, there's "Refresh list", which gives nothing...

Does it expects a specific store file structure? I primarily just have a bunch of `foo.com.gpg` files on my store root.

**Reply by msfjarvis** [reply]
> @msfjarvis just tried that and ended up the same: empty main page. All I did was to follow initial app walk-through, I've set a local password store path on my SD card, it asked for key access on OpenKeychain, which I selected, after that no passwords on front page, refresh of it asks for cloning repository first, `...` at top right offers "Synchronize repository" which gives a "Git url is not set!" error, there's also "Pull/Push from remote" which doesn't make sense for a bare _gitless_ local store, there's "Refresh list", which gives nothing...
> 
> Does it expects a specific store file structure? I primarily just have a bunch of `foo.com.gpg` files on my store root.

I tested this flow with our official testing repo at https://github.com/android-password-store/pass-test and it works fine, so I'm unsure what's going wrong on your device. If you are able to capture and provide device logs I'll be able to take a look next week.

**Reply by oblitum** [reply]
@msfjarvis there's a bug in the application, I think no log is needed. What's happening is that I put the keys in a subfolder of my SD-Card, it's in `/storage/emulated/0/Foo/pass`. When I navigate and select that folder for setting up "create local repository" (this is confusing terms for this, imo), I get empty password page because the app doesn't respect the directory structure, when I go to settings the directory is set to `/storage/emulated/0/pass`, with `Foo/` stripped, no matter what I try it always strips that. Checking my SD-Card root I notice the app has created an empty `pass/` there, with just a `.git` directory and `gpg-id` file. Deleting that and copying  `/storage/emulated/0/Foo/pass` to  `/storage/emulated/0/pass` cause the app to display the passwords. So it seems to be supporting stores in sd-card root solely.

**Reply by Skrilltrax** [reply]
@oblitum I'll take a look at the issue. Just to be sure, you've faced this issue in both the Play Store release and the snapshot?

**Reply by oblitum** [reply]
@Skrilltrax yup.

**Reply by Skrilltrax** [reply]
@oblitum I tried selecting a repo in a nested directory structure and the selection was fine. The issue that you mentioned happens when the folder path is not available in the app. In terms of UI, you need to open the folder where the passwords are stored and then press the button available below (Use this folder). If it's still an issue, please provide some logs and appropriate screenshots so that we can debug it further.

**Reply by oblitum** [reply]
> In terms of UI, you need to open the folder where the passwords are stored and then press the button available below (Use this folder).

It's exactly as I did in my test description above.

Does the app save log files when opting to enable logs (where?)? Otherwise I'll just use it with the bug as I'm not ok with attaching some debugger etc.

**Reply by oblitum** [reply]
I'd just like to add that on my previous tests the directory was at my device's memory, not an external sd-card in it. Could not set a whole path at all, I'm using from root dir to avoid the problem.
